### PR TITLE
opam: unbreak build process and add unzip to dependencies

### PR DESCRIPTION
**Pull request by jnbr**
In opam-1.2.2 there is a broken url in a Makefile, breaking the build process.
This gets fixed in post_extract.

Also, opam depends on unzip, so it is added as a dependency.

### CODE REVIEW

File: srcpkgs/opam/template [hunk 1]
```diff
@@ -15,3 +15,9 @@ distfiles="https://github.com/ocaml/${pkgname}/archive/${version}.tar.gz"
 checksum=3e4a05df6ff8deecba019d885ebe902eb933acb6e2fc7784ffee1ee14871e36a
 nocross=yes
 nopie=yes
+
+post_extract() {
+	# fix broken url in src_ext/Makefile
+	sed -i -e 's/http:\/\/ocaml-extlib.googlecode.com\/files\/extlib-1.5.3.tar.gz/http:\/\/storage.googleapis.com\/google-code-archive-downloads\/v2\/code.google.com\/ocaml-extlib\/extlib-1\.5\.3\.tar\.gz/' \
+	       $XBPS_BUILDDIR/${pkgname}-${version}/src_ext/Makefile
```
Comment (Vaelatern) [inline srcpkgs/opam/template]: Should be `src_ext/Makefile` instead of the `$XBPS_BUILDDIR/${pkgname}-${version}/......` stuff.

Comment (jnbr) [inline srcpkgs/opam/template]: Thank you, I will fix it.
Just messed up with git, so I will close this PR and reopen a clean new one.

**Comment by Vaelatern** [comment]
PLEASE DO NOT CLOSE AND REOPEN PRS. There is never a need to. Never.

**Comment by jnbr** [comment]
Sorry, won't happen again.

**Comment by Vaelatern** [comment]
It's right there in [CONTRIBUTING.md](https://github.com/voidlinux/void-packages/commit/c8c3099643251102bfb24352ba36c4d66ebbf733)

**Pull request closed without merging**